test/functional/services/saved_objects/fetch_saved_objects.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { writeFileSync } from 'fs';
+// @ts-ignore
+import { join } from 'path';
+// @ts-ignore
+import { ToolingLog } from '@kbn/dev-utils';
+import { SuperTest } from 'supertest';
+import { mkDir, finalDirAndFile, ndjsonToObj, mark } from './utils';
+// @ts-ignore
+import * as Either from '../../../../src/dev/code_coverage/ingest_coverage/either';
+
+const encoding = 'utf8';
+
+const writeUtf8 = { flag: 'w', encoding };
+
+const appendUtf8 = { flag: 'a', encoding };
+
+const defaultTypes = ['index-pattern', 'search', 'visualization', 'dashboard'];
+
+const isEmptyEither = (x: string) => (x === '' ? Either.left(x) : Either.right(x));
+
+const prokEither = (resp: any) => isEmptyEither(resp.text).map(ndjsonToObj);
+
+export const exportSavedObjects = (types = defaultTypes, excludeExportDetails = true) => async (
+  log: ToolingLog,
+  supertest: SuperTest<any>
+) =>
```

Comment:
Tre' and I discussed this a bit but I'm still not sure why we need to pass log and supertest to the saved objects functions here.  This is already importing supertest.  This seems different and a bit more complicated than other existing services.

Reply:
It's only importing the SuperTest type definition.  The supertest svc is what's being passed in here, not the type def.

Reply:
The log svc is passed in so we can log success or failure to the console.

Reply:
@LeeDr I can try just getting super test and the log in the index file, to simplify the api from the caller's point of view.